test/interactions/clickInteractionTests.ts

```diff
@@ -2,156 +2,186 @@
+    });
 
+    afterEach(function() {
+      if (this.currentTest.state === "passed") {
         svg.remove();
-      });
-
-      it("treats touch events same as click events for callbacks registered with onClick()", () => {
-        let callbackCalled = false;
-        let lastPoint: Plottable.Point;
-        let callback = function(point: Plottable.Point) {
-          callbackCalled = true;
-          lastPoint = point;
-        };
-        clickInteraction.onClick(callback);
-
-        TestMethods.triggerFakeTouchEvent("touchstart", component.content(), [{x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2}]);
-        TestMethods.triggerFakeTouchEvent("touchend", component.content(), [{x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2}]);
-        assert.isTrue(callbackCalled, "callback called on clicking Component without moving mouse");
-        assert.deepEqual(lastPoint, { x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2 }, "was passed correct point");
-
-        callbackCalled = false;
-        lastPoint = null;
-        TestMethods.triggerFakeTouchEvent("touchstart", component.content(), [{x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2}]);
-        TestMethods.triggerFakeTouchEvent("touchend", component.content(), [{x: SVG_WIDTH / 4, y: SVG_HEIGHT / 4}]);
-        assert.isTrue(callbackCalled, "callback called on clicking and releasing inside the Component");
-        assert.deepEqual(lastPoint, { x: SVG_WIDTH / 4, y: SVG_HEIGHT / 4 }, "was passed mouseup point");
-
-        callbackCalled = false;
-        TestMethods.triggerFakeTouchEvent("touchstart", component.content(), [{x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2}]);
-        TestMethods.triggerFakeTouchEvent("touchend", component.content(), [{x: SVG_WIDTH * 2, y: SVG_HEIGHT * 2}]);
-        assert.isFalse(callbackCalled, "callback not called if released outside Component");
-
-        callbackCalled = false;
-        TestMethods.triggerFakeTouchEvent("touchstart", component.content(), [{x: SVG_WIDTH * 2, y: SVG_HEIGHT * 2}]);
-        TestMethods.triggerFakeTouchEvent("touchend", component.content(), [{x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2}]);
-        assert.isFalse(callbackCalled, "callback not called if started outside Component");
-
-        TestMethods.triggerFakeTouchEvent("touchstart", component.content(), [{x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2}]);
-        TestMethods.triggerFakeTouchEvent("touchmove", component.content(), [{x: SVG_WIDTH * 2, y: SVG_HEIGHT * 2}]);
-        TestMethods.triggerFakeTouchEvent("touchend", component.content(), [{x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2}]);
-        assert.isTrue(callbackCalled, "callback called even if moved outside Component");
+      }
+    });
 
-        svg.remove();
+    type ClickTestCallback = {
+      lastPoint: Plottable.Point;
+      called: boolean;
+      reset: () => void;
+      (p: Plottable.Point): void;
+    }
+
+    function makeClickCallback() {
+      let callback = <ClickTestCallback> function(p?: Plottable.Point) {
+        callback.lastPoint = p;
```

Comment:
The tests pass, so undeniably it works. It works because it's only important that `callback` be defined at the time the method gets invoked. It should be equivalent to this:
```Javascript
var callback;
callback = function() { .. }
```